### PR TITLE
User model new

### PR DESCRIPTION
**Pull request by Rostammahabadi**
### Type of change made:
- [x] User model to add description (STRING), address (STRING), and image (STRING)
- [x] Subscription model update to add nextOrder (STRING)
### Detailed Description
- I added the migration for the user to add the fields
- I added the field to the model
- Added the mutations
- Added the type and resolvers
- Added the migration for the change to subscriptions
- Added the mutation to update the nextOrder field
- Added the field and model update
### Why is this change required? What problem does it solve?
ALL THE PROBLEMS
### Were there any challenges that arose while implementing this feature? If so, how were the addressed?
N/A
### Files added:
- [x] code/api/src/migrations
- [x] code/api/src/seeders

### CODE REVIEW

File: code/api/src/migrations/6-add_next_order_to_subscriptions.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    	})
+  	]);
+	},
+
+	down: function (queryInterface, Sequelize) {
+  	return Promise.all([
+      queryInterface.removeColumn('subscriptions', 'nextOrder')
+    ]);
+	}
+};
```
Comment (Rostammahabadi) [inline code/api/src/migrations/6-add_next_order_to_subscriptions.js:15]: Adds nextOrder to subscriptions table

File: code/api/src/modules/subscription/model.js [hunk 2]
```diff
@@ -8,6 +8,9 @@ module.exports = function(sequelize, DataTypes) {
     },
     crateId: {
       type: DataTypes.INTEGER
+    },
+    nextOrder: {
+      type: DataTypes.STRING
```
Comment (Rostammahabadi) [inline code/api/src/modules/subscription/model.js:13]: adds datatype for the table

File: code/api/src/modules/subscription/types.js [hunk 3]
```diff
@@ -15,8 +14,9 @@ const SubscriptionType = new GraphQLObjectType({
     user: { type: UserType },
     crate: { type: CrateType },
     createdAt: { type: GraphQLString },
-    updatedAt: { type: GraphQLString }
+    updatedAt: { type: GraphQLString },
```
Comment (Rostammahabadi) [inline code/api/src/modules/subscription/types.js:17]: Storing as a string, maybe for future iterations, we can change to type Date through the graphql-iso-date add-on

File: code/api/src/migrations/6-add_next_order_to_subscriptions.js (hunk 1 above)
Comment (EricLarson2020) [reply code/api/src/migrations/6-add_next_order_to_subscriptions.js:15]: Nice work here!

File: code/api/src/modules/subscription/model.js (hunk 2 above)
Comment (EricLarson2020) [reply code/api/src/modules/subscription/model.js:13]: great job!

File: code/api/src/migrations/6-add_next_order_to_subscriptions.js (hunk 1 above)
Comment (carflor) [reply code/api/src/migrations/6-add_next_order_to_subscriptions.js:15]: Nice use of Promise.all

File: code/api/src/modules/subscription/types.js (hunk 3 above)
Comment (carflor) [reply code/api/src/modules/subscription/types.js:17]: I have never used the graphql-iso-date but it sounds interesting

File: code/api/src/modules/subscription/mutations.js [hunk 4]
```diff
@@ -27,4 +26,24 @@ export const subscriptionRemove = {
...
   },
   resolve: remove
-}
\ No newline at end of file
+}
+
+// nextOrder Update
+export const nextOrderUpdate = {
+  type: SubscriptionType,
+  args: {
```
Comment (EricLarson2020) [inline code/api/src/modules/subscription/mutations.js:34]: so we need to pass this a nextOrder, a user, and a crate to update a nextOrder

File: code/api/src/modules/subscription/resolvers.js [hunk 5]
```diff
@@ -58,4 +58,16 @@ export async function remove(parentValue, { id }, { auth }) {
   } else {
     throw new Error('Access denied.')
   }
-}
\ No newline at end of file
+}
+
```
Comment (EricLarson2020) [inline code/api/src/modules/subscription/resolvers.js:62]: So all mutations require authentication before they are allowed

File: code/api/src/modules/subscription/types.js (hunk 3 above)
Comment (EricLarson2020) [reply code/api/src/modules/subscription/types.js:17]: Nice Work!

File: code/api/src/modules/user/resolvers.js [hunk 6]
```diff
@@ -58,6 +58,50 @@ export async function login(parentValue, { email, password }) {
   }
 }
 
+
+// Update Email
```
Comment (EricLarson2020) [inline code/api/src/modules/user/resolvers.js:62]: Nice work on this Rostam!

File: code/api/src/modules/subscription/mutations.js (hunk 4 above)
Comment (Rostammahabadi) [reply code/api/src/modules/subscription/mutations.js:34]: exactly! 😄

File: code/api/src/modules/subscription/resolvers.js (hunk 5 above)
Comment (Rostammahabadi) [reply code/api/src/modules/subscription/resolvers.js:62]: for the mutations that regard a specific user, yeah well want to authorize them before they can make any update ( i believe) i'm not sure how it works with the front end but i think it checks to see if they have a token stored in their cookies

Comment (stellakunzang) [reply code/api/src/modules/subscription/resolvers.js:62]: cool that makes sense

File: code/api/src/modules/subscription/types.js (hunk 3 above)
Comment (stellakunzang) [reply code/api/src/modules/subscription/types.js:17]: Good note. Maybe create an issue with a tag for future iteration so there is evidence that this is on the to do list? Ian suggested that last mod for things like this and it was really helpful.

File: code/api/src/modules/user/mutations.js [hunk 7]
```diff
@@ -37,4 +37,56 @@ export const userRemove = {
...
+    image: {
+      image: 'image',
+      type: GraphQLString
+    }
+  },
+  resolve: updateImageResolver
+}
+=======
+}
+>>>>>>> main
```
Comment (stellakunzang) [inline code/api/src/modules/user/mutations.js:92]: `HEAD` and `main` here are a little worrisome and often indicate future errors 🧐

File: code/api/src/seeders/4-subscriptions.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+      },
+      {
+        userId: 2,
+        crateId: 2,
+        nextOrder: "2012-01-02",
+        createdAt: new Date(),
+        updatedAt: new Date()
+      }
+    ], {})
+  },
```
Comment (stellakunzang) [inline code/api/src/seeders/4-subscriptions.js:21]: That's awesome that you made seeds! Thanks so much, everything here is so thorough.

**Pull request merged**